tests/extension/oneapi_sub_group_mask/sub_group_mask_set_api.cpp

```diff
@@ -16,18 +16,25 @@ using namespace sycl_cts;
 #ifdef SYCL_EXT_ONEAPI_SUB_GROUP_MASK
 
 struct check_result_set {
-  bool operator()(sycl::ext::oneapi::sub_group_mask &sub_group_mask,
+  bool operator()(sycl::ext::oneapi::sub_group_mask sub_group_mask,
                   const sycl::sub_group &sub_group) {
+    int shift = CHAR_BIT * sizeof(unsigned long) - sub_group_mask.size();
+    // sub_group_mask's size is expected to be greater than 0 and not to be
+    // greater than 32 in these tests but check is added to make sure of it
+    if (shift < 0 || sub_group_mask.size() == 0) return false;
```

Comment:
To be pedantic till the end, then the comment doesn't correspond the code. `shift < 0` doesn't match "sub_group_mask's size is expected not to be greater than 32" if sizeof(unsigned long) > 4. There are platforms where sizeof(unsigled long) == 8.

It might be easier to make the check straightforward and do exactly what comment says before computing the `shift` value. In addition to that, I would consider rephrasing the comment to something like "sub_group_mask's size must be in the range between 0 (excluded) and 32 (included)".

Reply:
Changed to straightforward check and changed comment.